components/SWTDetect.coffee

```diff
@@ -0,0 +1,43 @@
+noflo = require 'noflo'
+path = require 'path'
+utils = require '../utils'
+
+# @runtime noflo-nodejs
+# @name SWTDetect
+
+exports.getComponent = ->
+  c = new noflo.Component
+  c.icon = 'font'
+  c.description = 'Stroke Width Transform text detector'
+
+  c.inPorts.add 'canvas',
+    datatype: 'object'
+    description: 'Canvas of image to be detected'
+
+  c.outPorts.add 'out',
+    datatype: 'object'
+    description: 'Bounding boxes of detected text'
+  c.outPorts.add 'error',
+    datatype: 'object'
+    required: false
+
+  noflo.helpers.WirePattern c,
+    in: 'canvas'
+    out: ['out', 'error']
+    forwardGroups: true
+    async: true
+  , (canvas, groups, outPorts, callback) ->
+    utils.writeCanvasTempFile canvas, (err, tmpFile) ->
```

Comment:
Can you write some safety checks about input data here? Something like `canvas?.width?>0`